src/java/com/puppetlabs/http/client/impl/Metrics.java

```diff
@@ -0,0 +1,158 @@
+        return timers;
+    }
+
+    public static ArrayList<Timer.Context> startBytesReadTimers(MetricRegistry registry,
+                                                                HttpRequest request,
+                                                                String[] metricId) {
+        if (registry != null) {
+            ArrayList<Timer.Context> urlTimers = startBytesReadUrlTimers(registry, request);
+            ArrayList<Timer.Context> allTimers = new ArrayList<>(urlTimers);
+            if (metricId != null) {
+                ArrayList<Timer.Context> metricIdTimers =
+                        startBytesReadMetricIdTimers(registry, metricId);
+                allTimers.addAll(metricIdTimers);
+            }
+            return allTimers;
+        }
+        else {
+            return null;
+        }
+    }
+
+    public static Map<String, Timer> getClientMetrics(MetricRegistry metricRegistry){
+        if (metricRegistry != null) {
+            return metricRegistry.getTimers(new ClientMetricFilter.ClientFilter());
+        } else {
+            return null;
+        }
+    }
+
+    public static Map<String, Timer> getClientMetricsWithUrl(MetricRegistry metricRegistry,
+                                                             final String url,
+                                                             final MetricType metricType){
+        if (metricRegistry != null) {
+            return metricRegistry.getTimers(new ClientMetricFilter.UrlFilter(url, metricType));
+        } else {
+            return null;
+        }
+    }
+
+    public static Map<String, Timer> getClientMetricsWithUrlAndVerb(MetricRegistry metricRegistry,
+                                                                    final String url,
+                                                                    final String verb,
+                                                                    final MetricType metricType){
+        if (metricRegistry != null) {
+            return metricRegistry.getTimers(new ClientMetricFilter.UrlAndVerbFilter(url,verb, metricType));
+        } else {
+            return null;
+        }
+    }
+
+    public static Map<String, Timer> getClientMetricsWithMetricId(MetricRegistry metricRegistry,
+                                                                  final String[] metricId,
+                                                                  final MetricType metricType){
+        if (metricRegistry != null) {
+            return metricRegistry.getTimers(new ClientMetricFilter.MetricIdFilter(metricId, metricType));
+        } else {
+            return null;
+        }
+    }
+
+    public static Map<String, ClientMetricData> computeClientMetricsData(Map<String, Timer> timers){
+        Map<String, ClientMetricData> metricsData = new HashMap<>();
+        if (timers != null) {
+            for (SortedMap.Entry<String, Timer> entry : timers.entrySet()) {
```

Comment:
Should this just be a `Map.Entry` rather than the more specific `SortedMap.Entry`?